pkg/util/testing/wrappers.go

```diff
@@ -104,3 +105,50 @@ func (r *ReplicatedJobWrapper) SetReplicas(val int) *ReplicatedJobWrapper {
 func (r *ReplicatedJobWrapper) Obj() jobset.ReplicatedJob {
 	return r.ReplicatedJob
 }
+
+// JobWrapper wraps a Job.
+type JobWrapper struct {
+	batchv1.JobTemplateSpec
+}
+
+// MakeJob creates a wrapper for a Job.
+func MakeJob(name, ns string) *JobWrapper {
+	return &JobWrapper{
+		batchv1.JobTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: ns,
+			},
+			Spec: batchv1.JobSpec{},
+		},
+	}
+}
+
+// SetCompletionMode sets the value of job.spec.completionMode
+func (j *JobWrapper) SetCompletionMode(mode batchv1.CompletionMode) *JobWrapper {
+	j.Spec.CompletionMode = &mode
+	return j
+}
+
+// SetCompletions sets the value of job.spec.completions
+func (j *JobWrapper) SetCompletions(completions int) *JobWrapper {
+	j.Spec.Completions = pointer.Int32(int32(completions))
+	return j
+}
+
+// SetParallelism sets the value of job.spec.parallelism
+func (j *JobWrapper) SetParallelism(parallelism int) *JobWrapper {
+	j.Spec.Parallelism = pointer.Int32(int32(parallelism))
+	return j
+}
+
+// SetTEmplate sets the value of the pod template spec (job.spec.template)
+func (j *JobWrapper) SetTemplate(template corev1.PodTemplateSpec) *JobWrapper {
+	j.Spec.Template = template
+	return j
+}
```

Comment:
I would remove all those three functions, we don't have any logic that is directly impacted by those values, so just set parallelism and completion to some default value in MakeJob

Reply:
Makes sense, I wasn't sure if in the wrapper pattern it would be unexpected for the object to have some default values that the user didn't explicitly set. Anyway I finished refactoring this.

Reply:
sg, I don't think you uploaded a new commit yet.

Reply:
sorry about that, pushed it now.